src/actions/requests.rs

```diff
@@ -538,27 +527,56 @@ impl<'a> RequestAction<'a> for Deglob {
         let result = rustw_handle.join();
         let mut deglob_str = match result {
             Ok(Ok(s)) => s,
-            _ => {
-                return Err(());
-            }
+            Ok(Err(e)) => return Err(e),
+            Err(_) => return Err((ErrorCode::InternalError, "Internal thread paniced")),
         };
 
         // Handle multiple imports.
         if deglob_str.contains(',') {
             deglob_str = format!("{{{}}}", deglob_str);
         }
 
-        // Send a workspace edit to make the actual change.
-        // FIXME should handle the response
-        let output = serde_json::to_string(
-            &RequestMessage::new(out.provide_id(),
-                                 "workspace/applyEdit".to_owned(),
-                                 ApplyWorkspaceEditParams { edit: make_workspace_edit(ls_util::rls_to_location(&span), deglob_str) })
-        ).unwrap();
-        out.response(output);
+        Ok(DeglobResult{
+            location: ls_util::rls_to_location(&span),
+            new_text: deglob_str,
+        })
+    }
+}
 
-        // Nothing to actually send in the response.
-        Ok(Ack)
+impl<'a> Action<'a> for Deglob {
+    type Params = Location;
+    const METHOD: &'static str = "rustWorkspace/deglob";
+
+    fn new(_: &'a mut LsState) -> Self {
+        Deglob
+    }
+}
+
+impl<'a> RequestAction<'a> for Deglob {
```

Comment:
To be fair, I would just delete the `rustWorkspace/deglob` extension on RLS side completely.
It's custom and there's a very good chance that only rls-vscode extension implements it, whereas when we switch to using LSP code action and commands we actually improve the support. This is not a very often used feature, I'd say getting and using lightbulb suggestions is more ergonomic and we can remove the logic on the client side fairly easily, so I wouldn't worry about backwards-incompatibility, if that's the case here.